code/modules/mob/living/simple_animal/bot/floorbot.dm

```diff
@@ -278,18 +304,36 @@
 				anchored = 0
 				target = null
 	else
+		
 		var/turf/open/floor/F = target_turf
-		mode = BOT_REPAIRING
-		visible_message("<span class='notice'>[src] begins repairing the floor.</span>")
-		spawn(50)
-			if(mode == BOT_REPAIRING)
-				F.broken = 0
-				F.burnt = 0
-				F.ChangeTurf(/turf/open/floor/plasteel)
-				mode = BOT_IDLE
-				update_icon()
-				anchored = 0
-				target = null
+		if(replacetiles && F != tiletype.turf_type)
+			mode = BOT_REPAIRING
+			visible_message("<span class='notice'>[src] begins replacing the floor tiles.</span>")
+			spawn(50)
```

Comment:
Thanks, I'll add that. I just took the current check from the code that's currently in the floorbot code but this will improve it.